### PR TITLE
Remove one extra copy in block deletion

### PR DESCRIPTION
**Pull request by jiacheliu3**
### What changes are proposed in this pull request?

One copy over the block list is avoided in `DefaultBlockDeletionContext`.

### Why are the changes needed?

The copy in `DefaultBlockDeletionContext.close()` guards two things:
1. A copy creates a view over the `mBlocks`, so the listeners all see the same list of blocks.
2. A copy attempts to provide better thread-safety (although it's not a reliable guarantee)

Point 1 is actually not so meaningful. On `close()`, the context really should have all block IDs ready and no one is updating the list.
Point 2 has been addressed by https://github.com/Alluxio/alluxio/pull/12715

AFAIS the copy is unnecessary now.

### Does this PR introduce any user facing changes?

No.

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/master/block/BlockMaster.java [hunk 1]
```diff
@@ -120,7 +121,7 @@
    * @param blockIds a list of block ids to remove from Alluxio space
    * @param delete whether to delete blocks' metadata in Master
    */
-  void removeBlocks(List<Long> blockIds, boolean delete) throws UnavailableException;
+  void removeBlocks(Collection<Long> blockIds, boolean delete) throws UnavailableException;
```
Comment (jiacheliu3) [inline core/server/master/src/main/java/alluxio/master/block/BlockMaster.java:124]: Changed signature so the ConcurrentLinkedQueue from DefaultDeletionContext can be used. After all the caller just needs something to iterate.

File: core/server/master/src/main/java/alluxio/master/file/DefaultBlockDeletionContext.java [hunk 2]
```diff
@@ -32,7 +33,7 @@
    */
   public DefaultBlockDeletionContext(BlockDeletionListener... listeners) {
     mListeners = Arrays.asList(listeners);
-    mBlocks = new ConcurrentLinkedQueue();
+    mBlocks = new ConcurrentLinkedQueue<>();
```
Comment (jiacheliu3) [inline core/server/master/src/main/java/alluxio/master/file/DefaultBlockDeletionContext.java:35]: added a missing generic sign

**Review by ggezer — APPROVED** [review]

**Comment by jiacheliu3** [comment]
Failed test is irrelevant to the change
```
Error: 7.437 [ERROR] Failures: 
Error: 7.438 [ERROR] alluxio.client.fs.LoadMetadataIntegrationTest.repeatedDirectorySync
Error: 7.438 [ERROR]   Run 1: LoadMetadataIntegrationTest.repeatedDirectorySync:216->checkListStatus:514->checkFunctionCall:538 Expected to be take between 1500 and 1750.0. actual duration (ms): 1083
Error: 7.438 [ERROR]   Run 2: LoadMetadataIntegrationTest.repeatedDirectorySync:216->checkListStatus:514->checkFunctionCall:538 Expected to be take between 1500 and 1750.0. actual duration (ms): 1029
```

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**